### PR TITLE
Move DependencyCreation to the abstract-client

### PR DESCRIPTION
**Pull request by Buckram123**
### Checklist

- [ ] CI is green.
- [x] Changelog updated.

### CODE REVIEW

File: framework/packages/abstract-app/Cargo.toml [hunk 1]
```diff
@@ -11,7 +11,7 @@ crate-type = ["cdylib", "rlib"]
 
 
 [features]
-interface-macro = ["dep:abstract-interface", "dep:cw-orch"]
+interface-macro = ["dep:abstract-interface", "dep:cw-orch", "abstract-client"]
```
Comment (CyberHoward) [inline framework/packages/abstract-app/Cargo.toml:14]: Why does this need abstract-client?

Comment (Buckram123) [reply framework/packages/abstract-app/Cargo.toml:14]: > Why does this need abstract-client?

Because of `DependencyCreation` trait

Comment (Buckram123) [reply framework/packages/abstract-app/Cargo.toml:14]: Ah yeah, should be only for `test-utils` right

Comment (CyberHoward) [reply framework/packages/abstract-app/Cargo.toml:14]: I think it's only needed in `test-utils`

**Comment by Buckram123** [comment]
`abstract-interface` already exposed in `abstract-app` with `interface-macro` feature

**Pull request closed without merging**